packages/widget/src/components/amount-selector/amount-selector.ts

```diff
@@ -1,117 +1,141 @@
 import type { Resource } from '@buildwithsygma/sygma-sdk-core';
 import type { HTMLTemplateResult } from 'lit';
 import { html } from 'lit';
-import { customElement, property, query } from 'lit/decorators.js';
+import { customElement, property, query, state } from 'lit/decorators.js';
 import { ifDefined } from 'lit/directives/if-defined.js';
-import { map } from 'lit/directives/map.js';
 import { when } from 'lit/directives/when.js';
 import { BaseComponent } from '../base-component/base-component';
+import type { DropdownOption } from '../internal/dropdown/dropdown';
+import { networkIconsMap } from '../../assets';
 import { styles } from './styles';
 
 @customElement('sygma-resource-selector')
 export class AmountSelector extends BaseComponent {
   static styles = styles;
+
   @property({
     type: Array,
     hasChanged: (n, o) => n !== o
   })
   resources: Resource[] = [];
 
-  @property({
-    type: Boolean
-  })
+  @property({ type: Boolean })
   disabled = false;
 
-  @property({
-    type: String
-  })
+  @property({ type: String })
   accountBalance?: string;
 
-  @property({
-    type: String
-  })
+  @property({ type: String })
   preselectedToken?: string;
 
-  @property({
-    type: Number
-  })
+  @property({ type: Number })
   preselectedAmount?: number;
 
-  @property({
-    attribute: false
-  })
+  @property({ attribute: false })
   onResourceSelected?: (resource: Resource) => void;
```

Comment:
@mpetrunic do you mean to merge it in the controller, so that we have a method like this?
```js
// Old
  onResourceSelected = (resource: Resource): void => {
    console.log('resource', resource);
    this.selectedResource = resource;
  };

  onResourceAmountChange = (amount: number): void => {
    console.log('resource amount', amount);
    this.resourceAmount = amount;
  };
  
  // New
    onResourceSelected = (resource: Resource, amount: number): void => {
    console.log('resource', resource);
    this.selectedResource = resource;
    this.resourceAmount = amount;
  };
  
```

Reply:
callback prop should have `(resource: Resource, amount: number): void` interface, so that parent component receives both resource and amount selected^^